src/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/RandomNumberGenerator.cs

```diff
@@ -96,12 +97,113 @@ public static void Fill(Span<byte> data)
             RandomNumberGeneratorImplementation.FillSpan(data);
         }
 
+        public static int GetInt32(int fromInclusive, int toExclusive)
+        {
+            if (fromInclusive >= toExclusive)
+                throw new ArgumentException(SR.Argument_InvalidRandomRange);
+
+            // The total possible range is [0, 4,294,967,295).
+            long adjustedRange = (long)toExclusive - fromInclusive;
+
+            // If there is only one possible choice, nothing random will actually happen, so return
+            // the only possibility.
+            if (adjustedRange == 1)
+            {
+                return fromInclusive;
+            }
+
+            // Subtract one to account for zero being an actual possibility. We know for certain that
+            // the range fits in a UInt32 without overflowing.
+            uint maskedRange = (uint)adjustedRange - 1;
+
+            int leadingZeroCount = LeadingZeroCount(maskedRange);
+
+            // Create a mask for the bits that we care about for the range. The other bits will be
+            // masked away.
+            uint mask;
+
+            // If the LZCNT is zero create a full mask to avoid overflowing when shifting.
+            if (leadingZeroCount == 0)
+            {
+                mask = unchecked((uint)-1);
+            }
+            else
+            {
+                mask = (1U << 32 - leadingZeroCount) - 1;
+            }
+
+            // We only want to generate as many bytes as required to satisfy the mask to not apply
+            // undue pressure to the underlying random number generator.
+            int bytesRequired = (32 + 7 - leadingZeroCount) / 8;
+
+            Span<byte> valueResult = stackalloc byte[sizeof(uint)];
+            valueResult.Fill(0);
+            uint result;
+
+            do
+            {
+                Span<byte> randomFillSlice = valueResult.Slice(0, bytesRequired);
+                RandomNumberGeneratorImplementation.FillSpan(randomFillSlice);
+                result = mask & BinaryPrimitives.ReadUInt32LittleEndian(valueResult);
+            }
+            while (result > maskedRange);
+
+            return (int)(result + fromInclusive);
```

Comment:
This will spill to a `long` then back to an `int`. Recommend instead `return (int)result + fromInclusive`.